build_defs/build_defs.bzl

```diff
@@ -17,14 +18,14 @@ def _run_schema2pkg(name, src, out, language_name, language_flag, file_extension
         name = name,
         srcs = [src],
         outs = [out],
-        deps = ["//src/tools:schema2pkg_srcs"],
+        deps = ["@arcs//src/tools:schema2pkg_srcs", "@arcs//tools"],
         # TODO: generated header guard should contain whole workspace-relative
         # path to file.
-        cmd = "./tools/sigh schema2pkg " +
+        cmd = "$(location @arcs//tools) schema2pkg " +
               language_flag + " " +
-              "--outdir $(dirname {OUT}) " +
+              "--outdir $(dirname $PWD/{OUT}) " +
               "--outfile $(basename {OUT}) " +
-              "{SRC}",
+              "$PWD/{SRC}",
```

Comment:
The reason why this is prefixed by $PWD (absolutize the src path) is because $PWD is the execroot.  The problem with bazel and tools/sigh is that it always changes the CWD to the repo root, but SRC is relative to your workspace root (e.g. particles/PipeApps/Foo.arcs). But, if you have multiple workspaces, then files in an external workspace are symlinked into an 'external/' subdirectory under exec_root.  Thus, SRC = external/particles/PipeApps/Foo.arcs, and this will fail. 

By absolutizing the file, schema2kotlin Manifest.load() will work with relative imports, regardless of the CWD.

Reply:
Thanks for the explanation. Could you add a short comment to the code so that others who read the code later can see it too?